emukit/core/constraints.py

```diff
@@ -0,0 +1,73 @@
+import logging
+from typing import Callable, Optional
+
+import numpy as np
+
+_log = logging.getLogger(__name__)
+
+
+class IConstraint:
+    def evaluate(self, x: np.ndarray) -> np.ndarray:
+        """
+        :param x: Input locations to evaluate constraint at
+        :return: Numpy array of shape (n_input,) where an element will be 1 if the corresponding input satisfies the
+                 constraint and zero if the constraint is violated
+        """
+        raise NotImplementedError
+
+
+class LinearInequalityConstraint(IConstraint):
```

Comment:
Isn't that just NonlinearInequalityConstraint with g(x) = Ax ? you could simplify the code a bit with this

Reply:
The optimizer will treat linear and non-linear constraints differently so it needs to know whether a certain constraint is linear or non-linear and they have to be represented differently. I don't think it's possible to merge the two inequality constraints.

Setting g(x) = Ax is possible but you would get lots of warnings from the optimizer and would be inefficient.